Automated Smoketest: :x:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 96f7fe96c856742de72c38e48b4db4d26baacdb7
Build: success
ECL Watch:  Rebuilding Site
Lint error(s): 
eclwatch/DFUQueryWidget.js: line 362, col 50, Expected a break statement before default.
1 error
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! eclwatch@1.0.0 lint: `jshint --config ./.jshintrc ./eclwatch`
npm ERR! Exit status 2
npm ERR! 
npm ERR! Failed at the eclwatch@1.0.0 lint script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm ERR! A complete log of this run can be found in:
npm ERR!     /mnt/disk1/home/vamosax/.npm/_logs/2018-08-01T13_19_13_269Z-debug.log
npm ERR! Test failed.  See above for more details.

Lint warning(s): 
npm WARN Local package.json exists, but node_modules missing, did you mean to install?

Install hpccsystems-platform-community_7.0.0-trunk0.el7.x86_64.rpm
Creating group hpcc ...\x1b[32m [   OK    ]  \x1b[0m 
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |89 |89 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |804 |804 |0 |
|test (thor) |729 |729 |0 |
|test (roxie) |878 |878 |0 |


HPCC Stop: OK
HPCC Uninstall: OK

[Build error:](https://gist.github.com/8f8898674ee0bc309c145ff20dfc38f7)
Build log [Build log](https://gist.github.com/HPCCSmoketest/8f8898674ee0bc309c145ff20dfc38f7#file-result-18-08-01-09-18-24-log)

Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   25 sec (00:00:25) |  171 sec (00:02:51) |  129 sec (00:02:09) |   10 sec (00:00:10) |   32 sec (00:00:32) | 1282 sec (00:21:22) |   19 sec (00:00:19) | 1668 sec (00:27:48) |

--- a/esp/src/eclwatch/DFUQueryWidget.js
+++ b/esp/src/eclwatch/DFUQueryWidget.js
@@ -345,6 +345,27 @@ define([
 
             //  Implementation  ---
             getFilter: function () {
+                if (this.workunitsGrid){
+                    var retVal = this.filter.toObject();
+                    if (retVal.Sortby){
+                        switch (retVal.Sortby){
+                            case "Smallest":
+                                this.workunitsGrid.set ("sort", [{ attribute: "FileSize", "descending": false }]);
+                                break;
+                            case "Largest":
+                                this.workunitsGrid.set ("sort", [{ attribute: "FileSize", "descending": true }]);
+                                break;
+                            case "Oldest":
+                                this.workunitsGrid.set ("sort", [{ attribute: "Modified", "descending": false }]);
+                                break;
+                            case "Newest":
+                                /* falls through */
+                            default:
+                                this.workunitsGrid.set ("sort", [{ attribute: "Modified", "descending": true }]);
+                                break;
+                            }
+                        }
+                    }
                 var retVal = this.filter.toObject();
                 if (retVal.StartDate && retVal.FromTime) {
                     lang.mixin(retVal, {

--- a/esp/src/eclwatch/nls/hpcc.js
+++ b/esp/src/eclwatch/nls/hpcc.js
@@ -249,6 +249,7 @@ define({root:
     Finished: "Finished",
     FindNext: "Find Next",
     FindPrevious: "Find Previous",
+    FirstN: "First N",
     FirstName: "First Name",
     FirstNRows: "First N Rows",
     Fixed: "Fixed",

--- a/esp/src/eclwatch/templates/DFUQueryWidget.html
+++ b/esp/src/eclwatch/templates/DFUQueryWidget.html
@@ -159,6 +159,13 @@
                             <option value="Superfiles Only">${i18n.SuperfilesOnly}</option>
                             <option value="Not in Superfiles">${i18n.NotInSuperfiles}</option>
                         </select>
+                        <input id="${id}FirstN" title="${i18n.FirstN}:" name="FirstN" colspan="2" style="width:15%" data-dojo-props="trim: true, placeHolder:'-1'" data-dojo-type="dijit.form.TextBox" />
+                        <select id="${id}Sortby" title="" name="Sortby" colspan="2" data-dojo-type="dijit.form.Select">
+                            <option value="Newest" selected="selected">${i18n.Newest}</option>
+                            <option value="Oldest">${i18n.Oldest}</option>
+                            <option value="Smallest">${i18n.Smallest}</option>
+                            <option value="Largest">${i18n.Largest}</option>
+                        </select>
                         <input id="${id}FromDate" title="${i18n.FromDate}:" name="StartDate" data-dojo-props="trim: true, placeHolder:'7/28/2013'" data-dojo-type="dijit.form.DateTextBox" />
                         <input id="${id}FromTime" title="" name="FromTime" data-dojo-props="trim: true, placeHolder:'7:30 AM'" data-dojo-type="dijit.form.TimeTextBox" />
                         <input id="${id}ToDate" title="${i18n.ToDate}:" name="EndDate" data-dojo-props="trim: true, placeHolder:'7/28/2013'" data-dojo-type="dijit.form.DateTextBox" />

